### PR TITLE
build(deps): bump github.com/distribution/distribution/v3 from 3.0.0-alpha.1 to 3.0.0-beta.1

### PR DESCRIPTION
**Pull request by Wwwsylvia**
Fixing #788

### CODE REVIEW

**Review by TerryHowe — APPROVED** [review]
/lgtm

File: go.mod [hunk 1]
```diff
@@ -1,10 +1,12 @@
 module oras.land/oras-go
 
-go 1.22
+go 1.22.4
+
+toolchain go1.22.5
```
Comment (shizhMSFT) [inline go.mod:5]: Is this line required?

File: go.mod [hunk 2]
```diff
@@ -1,10 +1,12 @@
 module oras.land/oras-go
 
-go 1.22
+go 1.22.4
```
Comment (shizhMSFT) [inline go.mod:3]: Can it be `go 1.22.0`?

Comment (Wwwsylvia) [reply go.mod:3]: [github.com/distribution/distribution/v3](https://github.com/distribution/distribution) requires go version >= `1.22.4`.

File: go.mod (hunk 1 above)
Comment (Wwwsylvia) [reply go.mod:5]: It's generated by `go mod tidy`. Let me try to remove it and see what happens.

Comment (Wwwsylvia) [reply go.mod]: OK, I removed the toolchain and the build still passed.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**